src/v/net/connection_rate.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/gate.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/net/net.hh>
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/node_hash_map.h>
+
+#include <chrono>
+
+using namespace std::chrono_literals;
+
+namespace net {
+
+struct connection_rate_info {
+    int64_t max_connection_rate;
+    absl::flat_hash_map<ss::sstring, int64_t> ovverides;
+};
+
+struct connection_rate_counter {
+    explicit connection_rate_counter(int64_t max_rate) noexcept
+      : max_tokens(max_rate)
+      , one_token_time(std::max(1000 / max_tokens, 1l))
+      , rate(max_rate)
+      , last_update_time(ss::lowres_clock::now()) {}
+
+    const int64_t max_tokens;
+    const std::chrono::milliseconds one_token_time;
```

Comment:
what is this used for?  - what's the `sizeof` chrono::milliseconds? is it int64? if not I'd make it int64

Reply:
It is used for sleep in background fiber
and also to get number of tokens for adding

Reply:
wha'ts the size

Reply:
`sizeof(std::chrono::milliseconds)` = 8

Reply:
Ah great.